sys/suit/v4/handlers.c

```diff
@@ -0,0 +1,548 @@
+/*
+ * Copyright (C) 2019 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup     sys_suit_v4
+ * @{
+ *
+ * @file
+ * @brief       SUIT v4
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <inttypes.h>
+
+#include "suit/coap.h"
+#include "suit/conditions.h"
+#include "suit/v4/suit.h"
+#include "suit/v4/handlers.h"
+#include "suit/v4/policy.h"
+#include "suit/v4/suit.h"
+#include "riotboot/hdr.h"
+#include "riotboot/slot.h"
+#include "cbor.h"
+
+#include "log.h"
+
+#define HELLO_HANDLER_MAX_STRLEN 32
+
+static int _handle_command_sequence(suit_v4_manifest_t *manifest, CborValue *it,
+        suit_manifest_handler_t handler);
+static int _common_handler(suit_v4_manifest_t *manifest, int key, CborValue *it);
+static int _common_sequence_handler(suit_v4_manifest_t *manifest, int key, CborValue *it);
+
+static int _hello_handler(suit_v4_manifest_t *manifest, int key, CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+
+    if (cbor_value_is_text_string(it)) {
+        size_t len = HELLO_HANDLER_MAX_STRLEN;
+        char buf[HELLO_HANDLER_MAX_STRLEN];
+        cbor_value_copy_text_string(it, buf, &len, NULL);
+        return SUIT_OK;
+    }
+    else {
+        LOG_DEBUG("_hello_handler(): unexpected value type: %u\n", cbor_value_get_type(
+                   it));
+        return -1;
+    }
+}
+
+static int _validate_uuid(suit_v4_manifest_t *manifest, CborValue *it, uuid_t *uuid)
+{
+    (void)manifest;
+    uuid_t uuid_manifest;
+    char uuid_str[UUID_STR_LEN + 1];
+    char uuid_str2[UUID_STR_LEN + 1];
+    size_t len = sizeof(uuid_t);
+    cbor_value_copy_byte_string(it, (uint8_t*)&uuid_manifest, &len, NULL);
+    uuid_to_string(&uuid_manifest, uuid_str);
+    uuid_to_string(uuid, uuid_str2);
+    LOG_INFO("Comparing %s to %s from manifest\n", uuid_str2, uuid_str);
+    return uuid_equal(uuid, &uuid_manifest) ? 0 : -1;
+}
+
+static int _cond_vendor_handler(suit_v4_manifest_t *manifest, int key, CborValue *it)
+{
+    (void)key;
+    LOG_INFO("validating vendor ID\n");
+    int rc = _validate_uuid(manifest, it, suit_get_vendor_id());
+    if (rc == SUIT_OK) {
+        LOG_INFO("validating vendor ID: OK\n");
+        manifest->validated |= SUIT_VALIDATED_VENDOR;
+    }
+    return rc;
+}
+
+static int _cond_device_handler(suit_v4_manifest_t *manifest, int key, CborValue *it)
```

Comment:
This code is currently not being used, we aren't setting a `device_uuid` or adding the condition to the manifest generator. I would remove it from this PR.

Reply:
> This code is currently not being used, we aren't setting a `device_uuid` or adding the condition to the manifest generator. 

True

> I would remove it from this PR.

That'd cause work now and increase work when we do add support for device uuids. I'd rather keep it in.